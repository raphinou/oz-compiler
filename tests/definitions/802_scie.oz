% Scie test
local
   fun {Scie Xs Ys}
      fun {ScieLoop Xs Is}
        case Xs
        of nil then nil
        [] X|Xr then
          case Is
          of nil then
            X*Ys.1|{ScieLoop Xr Ys.2}
          [] I|Ir then
            X*I|{ScieLoop Xr Ir}
          end
        end
      end
   in
      {ScieLoop Xs Ys}
   end
in
   {Show {Scie [1 2 3 4 5 6] [10 100 1000]}}
end
