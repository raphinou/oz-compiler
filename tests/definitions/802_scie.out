[10 200 3000 40 500 6000]
